### PR TITLE
docs: add configuration examples to readme

### PR DESCRIPTION
**Pull request by claby2**
This PR primarily adds configuration examples to `README.md` and some general documentation improvements.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,9 +47,69 @@ For example, to show the working tree status of the repository:
...
+
+#### Implicit match
+
+If no `=>` operator is provided, it is assumed that the path given is both the `HOME` and `REPO` path.
+
+Symlink `${HOME}/.config/ambit/config.ambit -> ${AMBIT_REPO_PATH}/.config/ambit/config.ambit`:
+
+    .config/ambit/config.ambit;
+
+#### Variant Expression
```
Comment (virchau13) [inline README.md:82]: This does not actually explain what the variant expression does.

File: README.md [hunk 2]
```diff
@@ -34,9 +47,69 @@ For example, to show the working tree status of the repository:
...
+*   `${HOME}/.config/nvim/init.vim -> ${AMBIT_REPO_PATH}/.config/nvim/init.vim`
+
+<!---->
+
+    .config/[
+        bat/bat.conf,
+        nvim/init.vim
+    ];
+
+#### Match expressions
```
Comment (virchau13) [inline README.md:96]: This does not actually explain what a match expression does.

File: README.md [hunk 3]
```diff
@@ -34,9 +47,69 @@ For example, to show the working tree status of the repository:
...
+| AMBIT_CONFIG_PATH    | Path to configuration file.            | `${HOME}/.config/ambit/config.ambit` |
+| AMBIT_REPO_PATH      | Path to dotfile repository directory.  | `${HOME}/.config/ambit/repo`         |
+
 ## Configuration
 
-`ambit` will search for a configuration file located at `${HOME}/.config/ambit/config.ambit`.
+The purpose of the configuration file is to set the paths of the symlinks.
+A symlink is defined with two parts: an existing file relative to `AMBIT_REPO_PATH`, and its destination relative to the system's home directory.
+
+### Configuration examples
```
Comment (virchau13) [inline README.md:66]: These are not examples, they are more like "syntax".

File: README.md [hunk 4]
```diff
@@ -34,9 +47,69 @@ For example, to show the working tree status of the repository:
...
+        nvim/init.vim
+    ];
+
+#### Match expressions
+
+Conditionally symlink by os and host:
+
+    {os(linux): .Xresources};
+
+    {host(plamorg): .zshrc};
```
Comment (virchau13) [inline README.md:102]: You may want to have an example with only one layer of `{ }` and with multiple choices.

File: README.md [hunk 5]
```diff
@@ -34,9 +47,69 @@ For example, to show the working tree status of the repository:
...
+
+*   `${HOME}/.config/bat/bat.conf -> ${AMBIT_REPO_PATH}/.config/bat/bat.conf`
+*   `${HOME}/.config/nvim/init.vim -> ${AMBIT_REPO_PATH}/.config/nvim/init.vim`
+
+<!---->
+
+    .config/[
+        bat/bat.conf,
+        nvim/init.vim
+    ];
```
Comment (virchau13) [inline README.md:94]: You may want to have an example with nested variant expressions.

File: README.md [hunk 6]
```diff
@@ -1,22 +1,27 @@
...
+
 Dotfile manager written in Rust.
 
-(Planned) features:
+`ambit` symbolically links files from a git repository to your system.
+Files can be easily synced and are managed through a custom configuration file.
+
+Features:
 
+*   Custom configuration syntax
```
Comment (virchau13) [inline README.md:12]: Try promoting it a bit more: `Custom configuration syntax, specifically tailored towards dotfile managing` :)

File: README.md [hunk 7]
```diff
@@ -1,22 +1,27 @@
 # ambit
 
+[![Rust](https://github.com/plamorg/ambit/actions/workflows/rust.yml/badge.svg)](https://github.com/plamorg/ambit/actions/workflows/rust.yml)
```
Comment (virchau13) [inline README.md:3]: This should not be named `Rust` - try `CICD`, `CI`, or look at other projects for inspiration.

**Review by virchau13 — CHANGES_REQUESTED** [review]
You may want to also show that you can access files in the root by starting your paths with `/`.

**Comment by claby2** [comment]
> You may want to also show that you can access files in the root by starting your paths with `/`.

I don't think this is a good idea to include right now. This functionality is currently untested and also goes against the intended functionality.

File: README.md (hunk 7 above)
Comment (claby2) [reply README.md:3]: Changing this in the markdown would only change the alternate text; it won't impact what it looks like when rendered... But I will change this anyway.

File: README.md [hunk 8]
```diff
@@ -34,17 +55,82 @@ For example, to show the working tree status of the repository:
...
+
+If no `=>` operator is provided, it is assumed that the path given is both the `HOME` and `REPO` path.
+
+Symlink `${HOME}/.config/ambit/config.ambit -> ${AMBIT_REPO_PATH}/.config/ambit/config.ambit`:
+
+    .config/ambit/config.ambit;
+
+#### Variant Expression
+
+Variant expressions, denoted with brackets (`[]`), are a shorthand way to set multiple paths.
```
Comment (virchau13) [inline README.md:92]: You may want to write something like "a variant expression `[a, b, c]` expands out to a, b, and c"

File: README.md [hunk 9]
```diff
@@ -34,17 +55,82 @@ For example, to show the working tree status of the repository:
...
+
+Symlink:
```
Comment (virchau13) [inline README.md:94]: This should be changed to "This code can be shortened by using a variant expression..." or similar.

File: README.md [hunk 10]
```diff
@@ -34,17 +55,82 @@ For example, to show the working tree status of the repository:
...
+        bat/bat.conf,
+        nvim/init.vim,
+        kitty/[kitty.conf, theme.conf]
+    ];
+
+#### Match expressions
+
+Match expressions, denoted with braces (`{}`), allow to conditionally sync files based on a system's operating system and host.
+
+Conditionally symlink by os and host:
```
Comment (virchau13) [inline README.md:113]: Explain what this does by giving specific examples using OS & host.

File: README.md [hunk 11]
```diff
@@ -55,3 +141,12 @@ Building:
...
     $ cargo test
+
+Formatting:
+
+    $ cargo fmt
+
+Linting:
+
+    $ cargo check
+    $ cargo clippy --all-targets --all-features
```
Comment (virchau13) [inline README.md:152]: You may want to add that our automatic CI checks `fmt`, `clippy`, and `test`.

**Review by virchau13 — CHANGES_REQUESTED** [review]

File: README.md [hunk 12]
```diff
@@ -34,24 +55,105 @@ For example, to show the working tree status of the repository:
...
-## Installation
+### Configuration syntax examples
 
-Build the `ambit` binary from source:
+#### Basic match
 
-    $ git clone git@github.com:plamorg/ambit.git
-    $ cd ambit
-    $ cargo install --path .
+Symlink `${HOME}/host.txt -> ${AMBIT_REPO_PATH}/a/repo.txt`:
```
Comment (virchau13) [inline README.md:78]: The `->` should not be part of the monospaced font.
```suggestion
Symlink `${HOME}/host.txt` to `${AMBIT_REPO_PATH}/a/repo.txt`:
```

File: README.md [hunk 13]
```diff
@@ -34,24 +55,105 @@ For example, to show the working tree status of the repository:
...
+Variant expressions, denoted with brackets (`[]`), are a shorthand way to set multiple paths.
+
+For instance, the variant expression `[a, b, c]` would expand out to `a`, `b`, and `c`.
+
+Instead of manually writing out the paths, a variant expression can be used:
+
+*   `${HOME}/.config/bat/bat.conf -> ${AMBIT_REPO_PATH}/.config/bat/bat.conf`
+*   `${HOME}/.config/nvim/init.vim -> ${AMBIT_REPO_PATH}/.config/nvim/init.vim`
+*   `${HOME}/.config/kitty/kitty.conf -> ${AMBIT_REPO_PATH}/.config/kitty/kitty.conf`
+*   `${HOME}/.config/kitty/theme.conf -> ${AMBIT_REPO_PATH}/.config/kitty/theme.conf`
```
Comment (virchau13) [inline README.md:101]: Same with the `->`s here.

File: README.md [hunk 14]
```diff
@@ -34,24 +55,105 @@ For example, to show the working tree status of the repository:
...
+
+#### Implicit match
+
+If no `=>` operator is provided, it is assumed that the path given is both the `HOME` and `REPO` path.
+
+Symlink `${HOME}/.config/ambit/config.ambit -> ${AMBIT_REPO_PATH}/.config/ambit/config.ambit`:
+
+    .config/ambit/config.ambit;
+
+#### Variant Expression
```
Comment (virchau13) [inline README.md:90]:
```suggestion
#### Variant expressions
```

**Review by virchau13 — CHANGES_REQUESTED** [review]
You may want to also add somewhere that you can use backslashes to escape special characters, e.g. `.config/badly\ named\[dir\/\}/config`.

**Review by virchau13 — APPROVED** [review]

**Pull request merged**